### PR TITLE
Added authentication repository base

### PR DESCRIPTION
**Pull request by CalicoChloe**

### CODE REVIEW

File: app/src/main/java/com/example/pricingpal/model/modules/SupabaseModule.kt [hunk 1]
```diff
@@ -65,12 +82,12 @@ object SupabaseModule {
      * client for Supabase obtained from the SupabaseClient.
      *
      */
-    @Provides
-    @Singleton
-    fun provideSupabaseGoTrue(client: SupabaseClient): GoTrue {
+    //@Provides
```
Comment (AbdoulieJNJie98) [inline app/src/main/java/com/example/pricingpal/model/modules/SupabaseModule.kt]: Did you mean to keep this commented out code? Also, was it not necessary for the goTrue authorization framework?

File: app/src/main/java/com/example/pricingpal/model/modules/SupabaseModule.kt [hunk 2]
```diff
@@ -7,8 +7,10 @@ import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
 import io.github.jan.supabase.SupabaseClient
 import io.github.jan.supabase.createSupabaseClient
-import io.github.jan.supabase.gotrue.GoTrue
-import io.github.jan.supabase.gotrue.gotrue
+import io.github.jan.supabase.gotrue.Auth
+//import io.github.jan.supabase.gotrue.GoTrue
```
Comment (AbdoulieJNJie98) [inline app/src/main/java/com/example/pricingpal/model/modules/SupabaseModule.kt]: Did you mean to keep line 11 and 13?

Comment (CalicoChloe) [reply app/src/main/java/com/example/pricingpal/model/modules/SupabaseModule.kt]: No, I didn't. Thanks for catching that. I removed it!

File: app/src/main/java/com/example/pricingpal/model/modules/SupabaseModule.kt (hunk 1 above)
Comment (CalicoChloe) [reply app/src/main/java/com/example/pricingpal/model/modules/SupabaseModule.kt]: I went ahead and removed it. From what I can see, the new provideSupabaseAuth handles it fine. It's pulling from the same library (io.github.jan.supabase.gotrue), just in a way that aligns more cleanly with the examples in the documentation.

**Review by AbdoulieJNJie98 — COMMENTED** [review]
Looks good!

**Review by AbdoulieJNJie98 — APPROVED** [review]

**Pull request merged**